### PR TITLE
Added API1600 support for HCP

### PR DESCRIPTION
**Pull request by VenkateshRavula**
### Description
Added API1600 support for HCP

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [x] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] New endpoints supported are updated in the endpoints-support.md file.
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/shared_samples/hypervisor_cluster_profile.rb [hunk 1]
```diff
@@ -75,5 +79,7 @@
 puts "\nCompliance preview details are :\n#{cp}\n"
 
 # Deletes the created hypervisor cluster profile
-hcp.delete
+# Delete method accepts 2 optional arguments - soft_delete(boolean) and force(boolean)
```
Comment (chebroluharika) [inline examples/shared_samples/hypervisor_cluster_profile.rb:82]: Please mention that soft_delete is mandatory in 1600

File: examples/shared_samples/hypervisor_cluster_profile.rb [hunk 2]
```diff
@@ -75,5 +79,7 @@
...
+# The default values for both the arguments is "false"
+hcp.delete(true, true)
```
Comment (chebroluharika) [inline examples/shared_samples/hypervisor_cluster_profile.rb:84]: Please try if hcp.delete(soft_delete = true, force = true)  is working or not. If it is, change it accordingly.

File: lib/oneview-sdk/resource/api1600/c7000/hypervisor_cluster_profile.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+require_relative '../../api1200/c7000/hypervisor_cluster_profile'
+
+module OneviewSDK
+  module API1600
+    module C7000
+      # Hypervisor cluster profile resource implementation for API1600 C7000
+      class HypervisorClusterProfile < OneviewSDK::API1200::C7000::HypervisorClusterProfile
+
+        # Both softDelete and force are optional arguments but from API1600 softDelete is made mandatory argument
```
Comment (chebroluharika) [inline lib/oneview-sdk/resource/api1600/c7000/hypervisor_cluster_profile.rb:20]: Please mention that "softDelete is mandatory in API1600"

File: spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::API1200::Synergy::HypervisorClusterProfile do
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API1000::Synergy::HypervisorClusterProfile' do
+    expect(described_class).to be < OneviewSDK::API1000::Synergy::HypervisorClusterProfile
```
Comment (chebroluharika) [inline spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb:18]: I think it should inherit API1200::C7000 class. If so, please update description also accordingly.

File: spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+require 'spec_helper'
+
+RSpec.describe OneviewSDK::API1000::Synergy::HypervisorClusterProfile do
+  include_context 'shared context'
+
+  it 'inherits from OneviewSDK::API800::Synergy::HypervisorClusterProfile' do
+    expect(described_class).to be < OneviewSDK::API800::Synergy::HypervisorClusterProfile
```
Comment (chebroluharika) [inline spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb:18]: I think it should inherit API1000::C7000 class. If so, please update description also accordingly.

File: spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb (hunk 4 above)
Comment (VenkateshRavula) [reply spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb:18]: The respective lib file for API1200::Synergy file inherits from API1000::Synergy file, so mentioned the same in spec file.

File: spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb (hunk 5 above)
Comment (VenkateshRavula) [reply spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb:18]: The respective lib file for API1000::Synergy file inherits from API800::Synergy file, so mentioned the same in spec file.

File: spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb (hunk 4 above)
Comment (chebroluharika) [reply spec/unit/resource/api1200/synergy/hypervisor_cluster_profile_spec.rb:18]: okay

File: spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb (hunk 5 above)
Comment (chebroluharika) [reply spec/unit/resource/api1000/synergy/hypervisor_cluster_profile_spec.rb:18]: okay

**Pull request merged**